components/brave_wallet_ui/common/slices/api.slice.ts

```diff
@@ -1279,12 +1297,65 @@ export function createWalletApi () {
             : (balancesResult &&
                 Object.keys(balancesResult).map((tokenAddress) => ({
                   type: 'TokenBalancesForChainId',
-                  id:
-                    arg.coin === CoinTypes.ETH
-                      ? `${arg.address}-${arg.coin}-${arg.chainId}-${tokenAddress}`
-                      : `${arg.pubkey}-${arg.coin}-${arg.chainId}-${tokenAddress}`
+                  id: [arg.accountId.uniqueKey, arg.chainId, tokenAddress].join('-')
                 }))) || ['TokenBalancesForChainId']
       }),
+      getHardwareAccountDiscoveryBalance: query<
+        string,
+        GetHardwareAccountDiscoveryBalanceArg
+      >({
+        queryFn: async (
+          { coin, chainId, address },
+          { dispatch },
+          extraOptions,
+          baseQuery
+        ) => {
+          const { jsonRpcService } = baseQuery(undefined).data // apiProxy
```

Comment:
please wrap this function's logic in a try-catch. thrown any encountered errors and log/return the error message